internal/mode/static/state/graph/backend_refs.go

```diff
@@ -288,16 +307,39 @@ func getServiceAndPortFromRef(
 
 	svc, ok := services[svcNsName]
 	if !ok {
-		return svcNsName, v1.ServicePort{}, field.NotFound(refPath.Child("name"), ref.Name)
+		return svcNsName, v1.ServicePort{}, []v1.IPFamily{}, field.NotFound(refPath.Child("name"), ref.Name)
 	}
 
 	// safe to dereference port here because we already validated that the port is not nil in validateBackendRef.
 	svcPort, err := getServicePort(svc, int32(*ref.Port))
 	if err != nil {
-		return svcNsName, v1.ServicePort{}, err
+		return svcNsName, v1.ServicePort{}, []v1.IPFamily{}, err
 	}
 
-	return svcNsName, svcPort, nil
+	svcIPFamilies := svc.Spec.IPFamilies
+
+	return svcNsName, svcPort, svcIPFamilies, nil
+}
+
+func verifyIPFamily(npCfg *NginxProxy, svcIPFamily []v1.IPFamily) error {
+	if npCfg.Source == nil {
```

Comment:
npCfg can also be nil (in case it hasn't been configured in the GatewayClass). let's check for that too, otherwise the code could panic